### PR TITLE
ISPN-6436 Overriding the name of an indexed property is not supported with DSL query

### PR DESCRIPTION
**Pull request by anistor**
https://issues.jboss.org/browse/ISPN-6436
https://issues.jboss.org/browse/ISPN-6508

### CODE REVIEW

**Comment by gustavonalle** [comment]
There some familiar commits in this PR, should it be rebased on master instead?

**Comment by anistor** [comment]
I'll rebase. I thought I already did that...

**Comment by anistor** [comment]
Not sure why github.com UI shows more commits. Should be only 2. Pls fetch again.

**Comment by anistor** [comment]
Now it looks ok. Only 2 commits.

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/AcceptObjectFilter.java [hunk 1]
```diff
@@ -33,7 +33,11 @@ public ObjectFilter withParameters(Map<String, Object> namedParameters) {
       if (namedParameters == null) {
          throw new IllegalArgumentException("namedParameters argument cannot be null");
       }
-      //todo validate params
+      for (String paramName : getParameterNames()) {
+         if (namedParameters.get(paramName) == null) {
+            throw new IllegalArgumentException("Query parameter '" + paramName + "' was not set");
```
Comment (slaskawi) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/AcceptObjectFilter.java]: Probably we need some internationalization here.

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/ObjectFilterImpl.java [hunk 2]
```diff
@@ -185,7 +185,11 @@ public ObjectFilter withParameters(Map<String, Object> namedParameters) {
       if (namedParameters == null) {
          throw new IllegalArgumentException("namedParameters argument cannot be null");
       }
-      //todo validate params
+      for (String paramName : getParameterNames()) {
+         if (namedParameters.get(paramName) == null) {
+            throw new IllegalArgumentException("Query parameter '" + paramName + "' was not set");
```
Comment (slaskawi) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/ObjectFilterImpl.java]: And here

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/RejectObjectFilter.java [hunk 3]
```diff
@@ -23,7 +23,11 @@ public ObjectFilter withParameters(Map<String, Object> namedParameters) {
       if (namedParameters == null) {
          throw new IllegalArgumentException("namedParameters argument cannot be null");
       }
-      //todo validate params
+      for (String paramName : getParameterNames()) {
+         if (namedParameters.get(paramName) == null) {
+            throw new IllegalArgumentException("Query parameter '" + paramName + "' was not set");
```
Comment (slaskawi) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/RejectObjectFilter.java]: Here as well. It's the 3rd time I'm seeing this loop - would it be possible to extract it somehow?

**Comment by slaskawi** [comment]
Only some small picky issues. Otherwise looks ok to me.

**Comment by anistor** [comment]
Fixed all.

**Comment by anistor** [comment]
Updated.

**Comment by anistor** [comment]
Rebased.

**Comment by gustavonalle** [comment]
pulling

**Comment by gustavonalle** [comment]
merged, thanks @anistor !

**Pull request closed without merging**